### PR TITLE
Add script to validate expected setuptools_scm version

### PR DESCRIPTION
**Pull request by brettlangdon**

### CODE REVIEW

File: .github/workflows/build_deploy.yml [hunk 1]
```diff
@@ -151,6 +155,16 @@ jobs:
...
+        if: github.event.action == 'workflow_dispatch'
+        # Include all history and tags
+        with:
+          fetch-depth: 0
+      - name: Validate deploy version
+        if: github.event.action == 'workflow_dispatch'
+        run: |
+          ./scripts/validate-version "${{ github.event.inputs.expectedVersion }}"
+
       - uses: pypa/gh-action-pypi-publish@master
```
Comment (Kyle-Verhoog) [inline .github/workflows/build_deploy.yml:168]: Aside from your PR but i notice that we're pinned to master here which might leave us vulnerable to zero days if the github action is compromised

**Review by Kyle-Verhoog — APPROVED** [review]
lgtm :+1:

Could we test it by doing some other action (rather than deploying)

**Comment by brettlangdon** [comment]
> lgtm 👍
> 
> Could we test it by doing some other action (rather than deploying)

we could create a separate workflow that did all of these steps except the last one?

we'd have to merge it into `master` to be able to run manually, and then we'd just delete it. not opposed, but is this what you were thinking?

**Comment by Kyle-Verhoog** [comment]
> we'd have to merge it into master to be able to run manually, and then we'd just delete it. not opposed, but is this what you were thinking?

We could do it on one of our forks/test repos

**Review by Yun-Kim — APPROVED** [review]

**Comment by jd** [comment]
What is this solving?

**Pull request merged**